### PR TITLE
Prisma roads

### PR DESCRIPTION
**Pull request by ThomasBarrial**

### CODE REVIEW

File: src/api/projects.js [hunk 1]
```diff
@@ -1,60 +1,99 @@
...
+  const { id } = req.params;
+  try {
+    const project = await prisma.project.findUnique({
+      where: {
+        id,
+      },
+    });
+    res.status(200).json(project);
+  } catch (err) {
+    res.status(500).json(err);
```
Comment (MatthiasWanner) [inline src/api/projects.js:28]: 500 HTTP codes are using to server errors. 500 is "internal server error". Please use another code.

File: src/api/projects.js [hunk 2]
```diff
@@ -1,60 +1,99 @@
...
+    .findMany()
+    .then((projects) => {
+      res.status(200).json(projects);
+    })
+    .catch((error) => {
+      res.status(500).json(error);
+    });
 });
 
-router.get('/:id', (req, res) => {
```
Comment (MatthiasWanner) [inline src/api/projects.js:10]: Then/catch in /projects route, async/await in /projects/:id. I think we should choose and use same method in all the ressource.

File: src/api/projects.js [hunk 3]
```diff
@@ -1,60 +1,99 @@
...
 router.post('/', (req, res) => {
-  const init = projetsExample;
-  const newProjects = {
-    id: [...init].pop().id + 1,
-    ...req.body,
-  };
-  init.push(newProjects);
-  res.status(201).json(init[init.length - 1]);
+  const { name, description, code, companyId, taxation } = req.body;
+  console.log(name, description, code, companyId);
```
Comment (MatthiasWanner) [inline src/api/projects.js:34]: Delete the console.log when you want to merge a branch

File: src/api/projects.js [hunk 4]
```diff
@@ -1,60 +1,99 @@
...
+  prisma.project
```
Comment (MatthiasWanner) [inline src/api/projects.js:35]: Verify if required fields are in the body before requesting DB and send 400 "Bad request" if not.

File: src/api/projects.js [hunk 5]
```diff
@@ -1,60 +1,99 @@
...
+          },
+        },
+      },
+    })
+    .then((project) => {
+      res.status(201).json(project);
+    })
+    .catch((err) => {
+      console.log(err);
+      res.status(500).json(err);
```
Comment (MatthiasWanner) [inline src/api/projects.js:54]: Not 500

File: src/api/projects.js [hunk 6]
```diff
@@ -1,60 +1,99 @@
...
+        },
+      },
+      where: {
+        id,
+      },
+    });
+    res.status(200).json(project);
+  } catch (err) {
+    console.log(err);
+    res.status(500).json(err);
```
Comment (MatthiasWanner) [inline src/api/projects.js:82]: Not 500

File: src/api/projects.js [hunk 7]
```diff
@@ -1,60 +1,99 @@
...
+          },
+        },
+      },
+      where: {
+        id,
+      },
+    });
+    res.status(200).json(project);
+  } catch (err) {
+    console.log(err);
```
Comment (MatthiasWanner) [inline src/api/projects.js:81]: Delete console.log

File: src/api/projects.js [hunk 8]
```diff
@@ -1,60 +1,99 @@
...
+            id: companyId,
+          },
+        },
+      },
+    })
+    .then((project) => {
+      res.status(201).json(project);
+    })
+    .catch((err) => {
+      console.log(err);
```
Comment (MatthiasWanner) [inline src/api/projects.js:53]: Delete console.log

File: src/api/projects.js [hunk 9]
```diff
@@ -1,60 +1,99 @@
...
+  const { id } = req.params;
+  try {
+    await prisma.project.delete({
+      where: {
+        id,
+      },
+    });
+    res.sendStatus(204);
+  } catch (err) {
+    res.status(500).json(err);
```
Comment (MatthiasWanner) [inline src/api/projects.js:96]: Not 500

**Review by MatthiasWanner — CHANGES_REQUESTED** [review]

File: src/api/projects.js [hunk 10]
```diff
@@ -1,60 +1,114 @@
...
+            id: companyId,
+          },
+        },
+      },
+    })
+    .then((project) => {
+      res.status(201).json(project);
+    })
+    .catch((err) => {
+      res.status(404).json(err);
```
Comment (Karnak19) [inline src/api/projects.js:54]: use async await

**Review by Karnak19 — CHANGES_REQUESTED** [review]

File: src/api/projects.js (hunk 10 above)
Comment (ThomasBarrial) [reply src/api/projects.js:54]: Done

**Review by MatthiasWanner — APPROVED** [review]

**Review by MatthiasWanner — APPROVED** [review]

**Pull request merged**